src/core/apiLayer/productService.ts

```diff
@@ -0,0 +1,125 @@
+import { stringify } from "query-string";
+import { useEffect, useReducer } from "react";
+
+import { apiUrl } from "@temp/constants";
+
+const fetchProducts = (criteria: any, limit: number, offset: number) => {
```

Comment:
does it make sense to have stronger typing for criteria?

Reply:
yes, but it was my plan to implement that when we are adding the filter.

Reply:
Fixed